Index: local.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>## This file must *NOT* be checked into Version Control Systems,\r\n# as it contains information specific to your local configuration.\r\n#\r\n# Location of the SDK. This is only used by Gradle.\r\n# For customization when using a Version Control System, please read the\r\n# header note.\r\n#Tue Jul 25 10:48:31 EEST 2023\r\nsdk.dir=C\\:\\\\Users\\\\ovasilenko\\\\AppData\\\\Local\\\\Android\\\\Sdk\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/local.properties b/local.properties
--- a/local.properties	
+++ b/local.properties	
@@ -4,5 +4,5 @@
 # Location of the SDK. This is only used by Gradle.
 # For customization when using a Version Control System, please read the
 # header note.
-#Tue Jul 25 10:48:31 EEST 2023
-sdk.dir=C\:\\Users\\ovasilenko\\AppData\\Local\\Android\\Sdk
+#Tue Jul 25 21:28:48 EEST 2023
+sdk.dir=C\:\\Users\\oleg1\\AppData\\Local\\Android\\Sdk
